src/dscanner/analysis/vcall_in_ctor.d

```diff
@@ -5,388 +5,273 @@
-            else if (hasStatic || attr.attribute == tok!"private" || attr.attribute == tok!"package")
-            {
-                pushVirtual(false);
-                pop = true;
-                break;
-            }
-        }
-
-        // final class... final function
-        if ((d.classDeclaration || d.functionDeclaration) && hasFinal)
-            pushIsFinal(true);
-
-        d.accept(this);
-
-        if ((d.classDeclaration || d.functionDeclaration) && hasFinal)
-            popIsFinal;
-    }
-
-    override void visit(const(FunctionCallExpression) exp)
-    {
-        // nested function are not virtual
-        pushNestedFunc(true);
-        exp.accept(this);
-        popNestedFunc();
-    }
-
-    override void visit(const(UnaryExpression) exp)
-    {
-        if (isInCtor)
-        // get function identifier for a call, only for this member (so no ident chain)
-        if (const IdentifierOrTemplateInstance iot = safeAccess(exp)
-            .functionCallExpression.unaryExpression.primaryExpression.identifierOrTemplateInstance)
-        {
-            const Token t = iot.identifier;
-            if (t != tok!"")
-            {
-                _ctorCalls[$-1] ~= t;
-            }
-        }
-        exp.accept(this);
-    }
-
-    override void visit(const(FunctionDeclaration) d)
-    {
-        if (isInClass() && !isInNestedFunc() && !isFinal() && !d.templateParameters)
-        {
-            bool virtualOnce;
-            bool notVirtualOnce;
-
-            const bool hasAttribs = d.attributes !is null;
-            const bool hasStatic = hasAttribs ? d.attributes.canFind!(a => a.attribute.type == tok!"static") : false;
-
-            // handle "private", "public"... for this declaration
-            if (d.attributes) foreach (attr; d.attributes.retro)
-            {
-                if (!hasStatic &&
-                   (attr.attribute == tok!"public" || attr.attribute == tok!"protected"))
-                {
-                    if (!isVirtual)
-                    {
-                        virtualOnce = true;
-                        break;
-                    }
-                }
-                else if (hasStatic || attr.attribute == tok!"private" || attr.attribute == tok!"package")
-                {
-                    if (isVirtual)
-                    {
-                        notVirtualOnce = true;
-                        break;
-                    }
-                }
-            }
-
-            if (!isVirtual && virtualOnce)
-                _virtualMethods[$-1] ~= d.name;
-            else if (isVirtual && !virtualOnce)
-                _virtualMethods[$-1] ~= d.name;
-
-        }
-        d.accept(this);
-    }
+	alias visit = BaseAnalyzerDmd.visit;
+	mixin AnalyzerInfo!"vcall_in_ctor";
+
+	private enum string KEY = "dscanner.vcall_ctor";
+	private enum string MSG = "a virtual call inside a constructor may lead to unexpected results in the derived classes";
+
+	private static struct FuncContext
```

Comment:
It seems like you doubled the tab size. Why?

Reply:
I haven't doubled the tab size. I used dfmt to automatically format the file. It was done using tab characters with 4 spaces. Before only spaces were used. I think github shows a wider spacing to differentiate between normal spaces and tabs.

![image](https://github.com/user-attachments/assets/7e5ad918-d548-4fc6-9fdb-932ded6999c2)

Reply:
Then maybe try dfmt with a 4 spaces instead of 4 tabs?

Reply:
I'm using a single TAB character (1 char in bottom right corner of the screenshot) that has the standard size of 4 spaces

Reply:
Does D's standard style use spaces instead of tabs?

Reply:
In dmd yes, we have spaces. I think what we are seing here is how a tab is interpreted (default to 4 spaces on your machine, default to 8 on github). Just use spaces and let's not waste more time on this nonsense.